addons/spreadsheet/static/src/pivot/pivot_table.js

```diff
@@ -171,8 +180,91 @@ export class SpreadsheetPivotTable {
         return this._rows[index];
     }
 
-    getRowTitle() {
-        return this._rowTitle;
+    /**
+     * @returns {PivotCell[][]}
+     */
+    getPivotCells(includeTotal = true) {
+        const pivotHeight = this.getColHeadersHeight() + this.getRowHeight();
+        const pivotWidth = this.getColWidth() + (includeTotal ? 1 : 0);
+
+        const domainArray = [];
+        for (let col = 0; col < pivotWidth; col++) {
+            domainArray.push([]);
+            for (let row = 0; row < pivotHeight; row++) {
+                if (!includeTotal && row === pivotHeight - 1) {
+                    continue;
+                }
+                domainArray[col].push(this._getPivotCell(col, row, includeTotal));
+            }
+        }
+        return domainArray;
+    }
+
+    _isTotalRow(row) {
+        const maxIndent = Math.max(...this._rows.map((row) => row.indent));
+        return this._rows[row].indent !== maxIndent;
+    }
+
+    /**
+     * @returns {PivotCell}
+     */
+    _getPivotCell(col, row, includeTotal = true) {
+        const colHeadersHeight = this.getColHeadersHeight();
+        if (col === 0 && row === colHeadersHeight - 1) {
+            return { content: this._rowTitle, isHeader: true, style: HEADER_STYLE };
+        } else if (row <= colHeadersHeight - 1) {
+            const domain = this._getColHeaderDomain(col, row);
+            const style = row === colHeadersHeight - 1 ? MEASURE_STYLE : TOP_LEVEL_STYLE;
+            return { domain, isHeader: true, style };
+        } else if (col === 0) {
+            const rowIndex = row - colHeadersHeight;
+            const domain = this._getRowDomain(rowIndex);
+            const indent = this._rows[rowIndex].indent;
+            //TODO DISCUSS: isn't undefined style on header a ugly ?
+            const style = indent <= 1 ? TOP_LEVEL_STYLE : indent === 2 ? HEADER_STYLE : undefined;
+            return { domain, isHeader: true, style };
+        } else {
+            const rowIndex = row - colHeadersHeight;
+            if (!includeTotal && this._isTotalRow(rowIndex)) {
```

Comment:
I've 2 different indexing for the rows (index in rows and index in whole pivot (headers + row)). I've no good ieads for the name. `rowIndex/fullPivotRowIndex` ? idk

Reply:
I would go for `rowIndex/fullRowIndex` or `pivotRowIndex/fullPivotRowIndex`

Reply:
not sure but could you do with another offset?